apps/backend/src/utils/environment.py

```
@@ -0,0 +1,155 @@
+    "systemctl is-system-running >/dev/null 2>&1 && echo 'SYSTEMD' || echo 'NOT_SYSTEMD'"
+)
+
+
+@dataclass
+class EnvironmentInfo:
+    """Information about the detected environment"""
+
+    is_wsl: bool = False
+    is_container: bool = False
+    has_systemd: bool = False
+    detected_os: Optional[str] = None
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)